src/main/java/org/broadinstitute/hellbender/engine/ReadWindowWalker.java

```diff
@@ -0,0 +1,198 @@
+            windows.addAll(shardInterval(interval, windowSize, windowPadding, readsSource, dictionary));
+        }
+
+        return windows;
+    }
+
+    /**
+     * Shard a single interval into windows using the {@link #windowSize} and {@link #windowPadding} arguments
+     *
+     * @param interval interval to shard; must be on the contig according to the provided dictionary
+     * @param windowSize desired window size; intervals larger than this will be sharded into windows up to this size
+     * @param windowPadding desired window padding; each windowed interval will be padded on both sides by this number of bases
+     * @param readsSource data source for reads
+     * @param dictionary sequence dictionary for reads
+     * @return List of {@link ReadWindow} objects, sharded and padded as necessary
+     */
+    @VisibleForTesting
+    protected static List<ReadWindow> shardInterval( final SimpleInterval interval, final int windowSize, final int windowPadding, final ReadsDataSource readsSource, final SAMSequenceDictionary dictionary ) {
+        if ( ! IntervalUtils.intervalIsOnDictionaryContig(interval, dictionary) ) {
+            throw new IllegalArgumentException("Interval " + interval + " not within the bounds of a contig in the provided dictionary");
+        }
+
+        final List<ReadWindow> shards = new ArrayList<>();
+
+        int start = interval.getStart();
+
+        while ( start <= interval.getEnd() ) {
+            int end = Math.min(start + windowSize - 1, interval.getEnd());
+
+            final SimpleInterval nextWindow = new SimpleInterval(interval.getContig(), start, end);
+            final SimpleInterval nextWindowPadded = nextWindow.expandWithinContig(windowPadding, dictionary);
+            shards.add(new ReadWindow(nextWindow, nextWindowPadded, readsSource));
+
+            start += windowSize;
+        }
+
+        return shards;
+    }
+
+    /**
+     * Returns the read filter (simple or composite) that will be applied to the reads in each window.
+     *
+     * The default implementation uses the {@link org.broadinstitute.hellbender.engine.filters.WellformedReadFilter} filter with all default options,
+     * as well as the {@link ReadFilterLibrary#MAPPED} filter.
+     *
+     * Default implementation of {@link #traverse()} calls this method once before iterating
+     * over the reads and reuses the filter object to avoid object allocation. Nevertheless, keeping state in filter objects is strongly discouraged.
+     *
+     * Subclasses can override to provide their own filters
+     * Multiple filters can be composed by using {@link org.broadinstitute.hellbender.engine.filters.ReadFilter} composition methods.
+     */
+    public CountingReadFilter makeReadFilter(){
+        return new CountingReadFilter("Wellformed", new WellformedReadFilter(getHeaderForReads()))
+                .and(new CountingReadFilter("Mapped", ReadFilterLibrary.MAPPED));
+    }
+
+    @Override
+    public void traverse() {
```

Comment:
is this method intentionally left not final? If yes, describe what the responsibilities of subclasses are. Otherwise, make final